### PR TITLE
Add minikube page to Creating a Cluster section.

### PR DESCRIPTION
**Pull request by dlorenc**
You can see the staged version here: http://dlorenc.github.io/docs/getting-started-guides/minikube/

cc @pwittrock

### CODE REVIEW

File: docs/getting-started-guides/minikube.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+---
+
+* TOC
+{:toc}
+
+### Requirements
+
+Minikube requires that VT-x/AMD-v virtualization is enabled in BIOS.
```
Comment (pwittrock) [inline docs/getting-started-guides/minikube.md]: Link to Details / 1 line command  to determine if action is required?  Does this apply to Macs, or is this Linux only?

File: docs/getting-started-guides/minikube.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+### Test it out
+
+List the nodes in your cluster by running:
+
+```shell
+kubectl get nodes
+```
+
+Use the built-in Docker daemon with:
```
Comment (pwittrock) [inline docs/getting-started-guides/minikube.md]: Not clear what this means.  Would be helpful to give context and when you would need this.

File: docs/getting-started-guides/minikube.md (hunk 1 above)
Comment (pwittrock) [inline docs/getting-started-guides/minikube.md]: Also, add a brief introduction of the purporse.  `Minikube starts a single node kubernetes cluster locally for purposes of development and testing.`

**Comment by pwittrock** [comment]
Update the other local cluster solutions to say that minikube is the preferred solution.

File: docs/getting-started-guides/minikube.md (hunk 1 above)
Comment (dlorenc) [inline docs/getting-started-guides/minikube.md]: > Link to Details / 1 line command to determine if action is required? Does this apply to Macs, or is this Linux only?

Clarified.

Comment (dlorenc) [inline docs/getting-started-guides/minikube.md]: > Also, add a brief introduction of the purporse. Minikube starts a single node kubernetes cluster locally for purposes of development and testing.

Done.

**Comment by dlorenc** [comment]
> Not clear what this means. Would be helpful to give context and when you would need this.
> Add a line note

Is this helpful?

File: docs/getting-started-guides/minikube.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Minikube requires that VT-x/AMD-v virtualization is enabled in BIOS on all platforms.
+
+To check that this is enabled on Linux, run:
+
+```shell
+cat /proc/cpuinfo | grep 'vmx\|svm'
+```
+
+This command should output something if the setting is enabled.
```
Comment (pwittrock) [inline docs/getting-started-guides/minikube.md]: Great.  Seems to be enable on my mac by default.  Any idea what mac versions this is enabled on by default?

File: docs/getting-started-guides/docker.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 ---
 ---
 
+**Note: [Minikube](../minikube/) is the recommended method of running Kubernetes on your local machine.**
```
Comment (pwittrock) [inline docs/getting-started-guides/docker.md]: Clarify that this page is not Minikube.  e.g. `Stop.  This guide has been superseded by Minikube which is the recommended...`

**Comment by dlorenc** [comment]
> Great. Seems to be enable on my mac by default. Any idea what mac versions this is enabled on by default?
> Add a line note

I haven't seen anything with an exact version number, but I did see a few posts that say something like: "this should be enabled automatically on newer macs". I just added a note to point out that it's probably enabled.

**Comment by dlorenc** [comment]
Made the language more aggressive to point to minikube :)

File: docs/getting-started-guides/minikube.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
+---
+---
+
+* TOC
+{:toc}
+
+Minikube starts a single node kubernetes cluster locally for purposes of development and testing.
+
```
Comment (pwittrock) [inline docs/getting-started-guides/minikube.md]: Quick outline of capabilities + incapabilities would be helpful.

Supports:
* kube-dns for service location
* NodePort service Type
* Ingress?

Does not support:
* LoadBalancer service Type

Comment (dlorenc) [inline docs/getting-started-guides/minikube.md]: Done.

**Comment by pwittrock** [comment]
@johndmulhausen I am going to merge this because it is a huge improvement over our existing solutions.  We can make additional changes in a follow up.

**Pull request merged**